src/lib/routes/admin-api/tag-type.ts

```diff
@@ -18,30 +28,125 @@ class TagTypeController extends Controller {
+                }),
+            ],
+        });
+        this.route({
+            method: 'post',
+            path: '',
+            handler: this.createTagType,
+            permission: UPDATE_TAG_TYPE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'createTagType',
+                    responses: { 201: createResponseSchema('tagTypeSchema') },
+                    requestBody: createRequestSchema('tagTypeSchema'),
+                }),
+            ],
+        });
+        this.route({
+            method: 'post',
+            path: '/validate',
+            handler: this.validate,
+            permission: UPDATE_TAG_TYPE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'validateTagType',
+                    responses: {
+                        200: createResponseSchema('validateTagTypeSchema'),
+                    },
+                    requestBody: createRequestSchema('tagTypeSchema'),
+                }),
+            ],
+        });
+        this.route({
+            method: 'get',
+            path: '/:name',
+            handler: this.getTagType,
+            permission: NONE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'getTagType',
+                    responses: {
+                        200: createResponseSchema('tagTypeSchema'),
+                    },
+                }),
+            ],
+        });
+        this.route({
+            method: 'put',
+            path: '/:name',
+            handler: this.updateTagType,
+            permission: UPDATE_TAG_TYPE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'updateTagType',
+                    responses: {
+                        200: emptyResponse,
+                    },
+                    requestBody: createRequestSchema('updateTagTypeSchema'),
+                }),
+            ],
+        });
+        this.route({
+            method: 'delete',
+            path: '/:name',
+            handler: this.deleteTagType,
+            permission: DELETE_TAG_TYPE,
+            middleware: [
+                openApiService.validPath({
+                    tags: ['admin'],
+                    operationId: 'deleteTagType',
+                    responses: {
+                        200: emptyResponse,
+                    },
+                }),
+            ],
+        });
     }
 
-    async getTagTypes(req: Request, res: Response): Promise<void> {
+    async getTagTypes(
+        req: Request,
+        res: Response<TagTypesSchema>,
+    ): Promise<void> {
         const tagTypes = await this.tagTypeService.getAll();
         res.json({ version, tagTypes });
```

Comment:
Could use `openApiService.respondWithValidation` here for runtime response validation.